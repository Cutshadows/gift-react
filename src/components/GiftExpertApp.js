import React, { Fragment, useState } from 'react';
import AddCategory from './AddCategory';
import GiftGrid from './GifGrid';

const GiftExpertApp = ({defaultCategories=[]}) => {

    //const categories=['One Punch', 'Samurai X', 'Dragon Ball'];
    // const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategories);

    // const handledAdd=()=>{
    //     setCategories(['Naruto', ...categories]); //insertar primero en la entrada
    //     //setCategories(cats=>[...cats, 'Naruto']); //Otra forma de insertar en el estado
    // }
    return ( 
        <Fragment>
            
            <h2>GiftExpertApp</h2>
            <hr/>
            <AddCategory setCategories={setCategories} />
           {/*<button onClick={handledAdd}>Agregar</button>*/}
            <h2>
                {
                   categories
                        .map(category =>(
                            <GiftGrid
                                key={category}
                                category={category} />
                            )
                        ) 
                }
            </h2>
        </Fragment>
     );
}
 
export default GiftExpertApp;