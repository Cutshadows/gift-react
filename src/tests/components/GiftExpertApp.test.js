import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GiftExpertApp from '../../components/GiftExpertApp';


describe('Llevaremos a cabo las pruebas de <GiftExpertApp />', () => {
    

    test('deberia hacer match con GiftExpertApp', () => {
        const wrapper=shallow(<GiftExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should show a list of categories ', () => {
        const categories=['One Punch', 'Dragon Ball'];
        const wrapper=shallow(<GiftExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GiftGrid').length).toBe(categories.length);
    })
    
    
})
