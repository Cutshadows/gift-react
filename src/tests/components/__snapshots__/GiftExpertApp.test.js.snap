// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Llevaremos a cabo las pruebas de <GiftExpertApp /> deberia hacer match con GiftExpertApp 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <h2 />
</Fragment>
`;

exports[`Llevaremos a cabo las pruebas de <GiftExpertApp /> should show a list of categories  1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <h2>
    <GiftGrid
      category="One Punch"
      key="One Punch"
    />
    <GiftGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </h2>
</Fragment>
`;
